ui/articles-aem-renderer/src/main/java/org/cru/godtools/articles/aem/service/support/ArticleParser.java

```diff
@@ -0,0 +1,233 @@
+package org.cru.godtools.articles.aem.service.support;
+
+import org.cru.godtools.articles.aem.model.Article;
+import org.cru.godtools.articles.aem.model.Attachment;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+
+import timber.log.Timber;
+
+/**
+ * This class handles parsing any AEM json calls into DOA objects.
+ */
+public class ArticleParser {
+
+    public static final String ARTICLE_LIST_KEY = "article_list_key";
+
+    public static final String ATTACHMENT_LIST_KEY = "attachment_list_key";
+
+    private JSONObject articleJSON;
+
+    private List<Attachment> attachmentList = new ArrayList<>();
+
+    private List<Article> articleList = new ArrayList<>();
+
+    private static final String CREATED_TAG = "jcr:created";
+    private static final String CONTENT_TAG = "jcr:content";
+    private static final String LAST_MODIFIED_TAG = "cq:lastModified";
+    private static final String UUID_TAG = "jcr:uuid";
+    private static final String TITLE_TAG = "jcr:title";
+    private static final String ROOT_TAG = "root";
+    private static final String FILE_TAG = "fileReference";
+    private static final String BASE_URL = "https://stage.cru.org";
+
+    /**
+     * Constructor
+     * @param jsonObject = the Object to be parsed.
+     */
+    public ArticleParser(JSONObject jsonObject) {
+        this.articleJSON = jsonObject;
+    }
+
+    //region Public Executor
+    /**
+     * This execute function allows you to change the initial jsonObject and run the execution.
+     *
+     * use <code>ARTICLE_LIST_KEY</code> and <code>ATTACHMENT_LIST_KEY</code> to access the Collections
+     * out of the HashMaps
+     *
+     * @param jsonObject = the object to be parsed.
+     * @return = return a list of <code>Article</code> and <code>Attachments</code> in HashMap
+     */
+    public HashMap<String, Object> execute(JSONObject jsonObject) {
```

Comment:
I don't think you ever use this version of `execute`

Reply:
from an overall class design perspective, this class has potential for data leakage if the same Parser object is used multiple times due to how the instance variables are used. There can be the intention that it will never be used in that way, but a developer a couple years down the road may not understand that constraint and mis-use the class.

Using instance variables is fine for tracking state, but I would make it so that `articleJSON` is final and only set in the constructor, and ensure you only parse the json one time when creating the object.

Reply:
Would it make more since to make execute a static method that takes in a local JsonObject, and use the local object to parse?